src/endpoints/pagination/iter.rs

```diff
@@ -1,14 +1,15 @@
 //! Synchronous implementation of automatic pagination requests.
 
-use crate::client::{ClientError, ClientResult};
-use crate::model::Page;
+use crate::{model::Page, ClientError, ClientResult};
 
 /// Alias for `Iterator<Item = T>`, since sync mode is enabled.
-pub trait Paginator<T>: Iterator<Item = T> {}
-impl<T, I: Iterator<Item = T>> Paginator<T> for I {}
+pub type Paginator<'a, T> = Box<dyn Iterator<Item = T> + 'a>;
```

Comment:
I don't totally get it now why would we need to declare type `pub type Paginator<'a, T> = Box<dyn Iterator<Item = T> + 'a>;` instead of the previous one ?
```rust
pub trait Paginator<T>: Iterator<Item = T> {}
impl<T, I: Iterator<Item = T>> Paginator<T> for I {}
```

Because we need some kinds of dynamic dispatch?

Reply:
A trait-based implementation of these pagination methods is only possible if we use `-> Box<dyn Paginator>` instead of  `-> impl Paginator`. The compiler will complain about it the same way we can't have `-> Future<...>`, for which we use `async_trait`. This is technically temporary and will be fixed in the future, but I'd give it *at least* a year until that happens, really. So we are basically forced to use `Box + dyn` if we want it to work for now. 

This is sub-optimal because there's a slight runtime overhead in comparison to generics, but on the other hand there should be less code bloat due to generics (not that either of these really matter, IMO).